test/unit/storage_test.rb

```diff
@@ -138,12 +185,11 @@ def assert_sentinel_connector(client)
     assert_instance_of Redis::Client::Connector::Sentinel, connector
   end
 
-  def assert_client_config(client, host: nil, port: nil, url: nil, sentinels: nil)
-    raise "bad usage of #{__method__}" unless host || port || url
-    assert_equal client.port, port if port
-    assert_equal client.host, host if host
-    assert_equal client.options[:url], url if url
-    assert_equal client.options[:sentinels], sentinels if sentinels
+  def assert_client_config(client, opts)
+    raise "bad usage of #{__method__}" unless opts[:host] || opts[:port] || opts[:url]
+    opts.each do |k, v|
+      assert_equal client.options[k], v if v
```

Comment:
why the `if v`? this is under the caller's control, so they might want to assert a specific options is nil? the original `if v` comes from the keyword arguments being mandatory (so the key always exists), but it's not the case with a hash.

Reply:
agree.

Done